src/esbmc/esbmc_parseoptions.cpp

```diff
@@ -1866,8 +1879,15 @@ bool esbmc_parseoptionst::process_goto_program(
 
       std::string filename = cmdline.args[0];
       goto_coveraget tmp(ns, goto_functions, filename);
-      tmp.replace_all_asserts_to_guard(gen_true_expr());
+
+      // if we do not want to count the guard in the assertions
+      if (cmdline.isset("no-cov-asserts"))
+        tmp.replace_all_asserts_to_guard(gen_true_expr());
       tmp.gen_cond_cov();
+
+      // redo conversion to remove_sideeffect
```

Comment:
why?

Reply:
Because we deliberately skip some of the sideeffects removal process when generating the Goto program. This is to keep the condition/guards format and avoid introducing auxiliary variables, which will affect the coverage calculation.

The documentation can be find here:
https://github.com/esbmc/esbmc/blob/f5057afb207cac2cb61b9be1b3ed2354366de96e/src/goto-programs/goto_coverage_rm.cpp#L3C1-L6C43